### PR TITLE
Keycloak-integration - integration to the cart service.

### PR DESCRIPTION
**Pull request by xcrou89**

### CODE REVIEW

File: src/main/java/com/stakater/nordmart/CartServiceApplication.java [hunk 1]
```diff
@@ -20,6 +22,11 @@ public static void main(String[] args)
         SpringApplication.run(CartServiceApplication.class, args);
     }
 
+    @Bean
+    public KeycloakSpringBootConfigResolver keycloakConfigResolver() {
+        return new KeycloakSpringBootConfigResolver();
+    }
+
```
Comment (ahmedwaleedmalik) [inline src/main/java/com/stakater/nordmart/CartServiceApplication.java]: Move this to a separate class `KeycloakConfiguration`

```
@org.springframework.context.annotation.Configuration
public class KeycloakConfiguration
{
 //your code for bean here
}
```

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -1,4 +1,15 @@
...
-application.mode=dev
\ No newline at end of file
+application.mode=dev
+
+keycloak.realm = ${KEYCLOAK_REALM}
+keycloak.auth-server-url = ${KEYCLOAK_URL}
+keycloak.ssl-required = external
+keycloak.resource = ${KEYCLOAK_CLIENT_BACKEND}
+keycloak.bearer-only = true
+keycloak.credentials.secret = ${KEYCLOAK_CLIENT_SECRET_KEY}
```
Comment (ahmedwaleedmalik) [inline src/main/resources/application.properties]: Provide default values where possible like

```
keycloak.realm = ${KEYCLOAK_REALM:nordmart}
```

File: deployment/values.yaml [hunk 3]
```diff
@@ -8,6 +8,14 @@ deployment:
   env:
   - name: CATALOG_ENDPOINT
     value: "catalog.nordmart-store:8080"
+  - name: KEYCLOAK_REALM
+    value: "nordmart"
+  - name: KEYCLOAK_URL
+    value: "http://192.168.99.100:8180/auth/"
```
Comment (ahmedwaleedmalik) [inline deployment/values.yaml]: replace value with `"https://keycloak-security.DOMAIN:8180/auth/"`

File: deployment/values.yaml [hunk 4]
```diff
@@ -8,6 +8,14 @@ deployment:
...
+  - name: KEYCLOAK_CLIENT_BACKEND
+    value: "stakater-nordmart-backend"
+  - name: KEYCLOAK_CLIENT_SECRET_KEY
+    value: "7672aea2-81ff-4aa2-affe-daf3bad10334"
```
Comment (ahmedwaleedmalik) [inline deployment/values.yaml]: Use `<concealed>`

**Review by ahmedwaleedmalik — CHANGES_REQUESTED** [review]

File: deployment/values.yaml (hunk 3 above)
Comment (xcrou89) [reply deployment/values.yaml]: Done.

File: deployment/values.yaml (hunk 4 above)
Comment (xcrou89) [reply deployment/values.yaml]: Done.

File: src/main/java/com/stakater/nordmart/CartServiceApplication.java (hunk 1 above)
Comment (xcrou89) [reply src/main/java/com/stakater/nordmart/CartServiceApplication.java]: Done.

File: src/main/resources/application.properties (hunk 2 above)
Comment (xcrou89) [reply src/main/resources/application.properties]: Done.

**Comment by stakater-user** [comment]
@xcrou89 Yikes! You better fix it before anyone else finds out! [Build 1](https://jenkins.release.stakater.com/job/nordmart-apps/job/stakater-nordmart-cart/job/PR-26/1/) has Failed!

**Comment by stakater-user** [comment]
@xcrou89 Image is available for testing. `docker pull docker.delivery.stakater.com:443/stakater-lab/cart:0.0.0-PR-26-2-SNAPSHOT`

**Comment by stakater-user** [comment]
@xcrou89 Image is available for testing. `docker pull docker.delivery.stakater.com:443/stakater-lab/cart:0.0.0-PR-26-3-SNAPSHOT`

**Comment by stakater-user** [comment]
@xcrou89 Image is available for testing. `docker pull docker.delivery.stakater.com:443/stakater-lab/cart:0.0.0-PR-26-4-SNAPSHOT`

**Comment by stakater-user** [comment]
@xcrou89 Image is available for testing. `docker pull docker.delivery.stakater.com:443/stakater-lab/cart:0.0.0-PR-26-5-SNAPSHOT`

**Comment by stakater-user** [comment]
@xcrou89 Image is available for testing. `docker pull docker.delivery.stakater.com:443/stakater-lab/cart:0.0.0-PR-26-9-SNAPSHOT`

File: README.md [hunk 5]
```diff
@@ -4,6 +4,17 @@
 
 A maven spring boot cart application.
 
+## Configurations
+
+Environment variables can be provided to configure the cart service.
+
+**Keycloak configurations**
+* CATALOG_ENDPOINT - Endpoint of the catalog endpoint
```
Comment (ahmedwaleedmalik) [inline README.md:11]: `CATALOG_ENDPOINT - Endpoint of the catalog endpoint` this is not part of keycloak configuration

**Review by ahmedwaleedmalik — CHANGES_REQUESTED** [review]

File: README.md (hunk 5 above)
Comment (xcrou89) [reply README.md:11]: Already move out of the Keycloak configurations part.

**Review by ahmedwaleedmalik — APPROVED** [review]

**Comment by stakater-user** [comment]
@xcrou89 Image is available for testing. `docker pull docker.delivery.stakater.com:443/stakater-lab/cart:0.0.0-PR-26-10-SNAPSHOT`

**Pull request merged**